packages/app/src/app/pages/Sandbox/Editor/Content/Preview/index.tsx

```diff
@@ -9,198 +11,249 @@ import getTemplate from '@codesandbox/common/lib/templates';
-  store: any;
-  signals: any;
   options: { url?: string };
-  reaction: any;
 };
 
-type State = {
-  running: boolean;
+const useForceUpdate = () => {
+  const [callback, setCallback] = useState(() => () => {});
+  const forceUpdate = useCallback(
+    cb => {
+      setCallback(() => cb);
+    },
+    [setCallback]
+  );
+
+  useEffect(() => {
+    if (_isFunction(callback)) {
+      callback();
+    }
+  }, [callback]);
+
+  return forceUpdate;
 };
 
-class PreviewComponent extends Component<Props, State> {
-  state: State = {
-    running: !this.props.runOnClick,
-  };
-
-  onPreviewInitialized = preview => {
-    const disposeHandleProjectViewChange = this.props.reaction(
-      ({ editor }) => editor.isInProjectView,
-      this.handleProjectView.bind(this, preview)
-    );
-    const disposeHandleForcedRenders = this.props.reaction(
-      ({ editor }) => editor.forceRender,
-      this.handleExecuteCode.bind(this, preview)
-    );
-    const disposeHandleExternalResources = this.props.reaction(
-      ({ editor }) => editor.currentSandbox.externalResources.length,
-      this.handleExecuteCode.bind(this, preview)
-    );
-    const disposeHandleModuleSyncedChange = this.props.reaction(
-      ({ editor }) => editor.isAllModulesSynced,
-      this.handleModuleSyncedChange.bind(this, preview)
-    );
-    const disposeHandleCodeChange = this.props.reaction(
-      ({ editor }) => String(editor.currentSandbox.modules.map(m => m.code)),
-      () => {
-        this.handleCodeChange(preview);
-      }
-    );
-    const disposeHandleModuleChange = this.props.reaction(
-      ({ editor }) => editor.currentModule,
-      () => {
-        if (!this.props.store.editor.isInProjectView) {
-          this.handleCodeChange(preview);
-        }
-      }
-    );
-    const disposeHandleStructureChange = this.props.reaction(
-      this.detectStructureChange,
-      this.handleStructureChange.bind(this, preview)
-    );
-    const disposeDependenciesHandler = this.props.reaction(
-      ({ editor }) =>
-        editor.currentSandbox.npmDependencies.keys
-          ? editor.currentSandbox.npmDependencies.keys().length
-          : Object.keys(editor.currentSandbox.npmDependencies),
-      this.handleDependenciesChange.bind(this, preview)
-    );
-
-    return () => {
-      disposeHandleModuleChange();
-      disposeHandleProjectViewChange();
-      disposeHandleForcedRenders();
-      disposeHandleExternalResources();
-      disposeHandleModuleSyncedChange();
-      disposeHandleCodeChange();
-      disposeHandleStructureChange();
-      disposeDependenciesHandler();
-    };
-  };
-
-  detectStructureChange = ({ editor }) => {
-    const sandbox = editor.currentSandbox;
-
-    return String(
-      sandbox.modules
-        .map(module => module.directoryShortid + module.title)
-        .concat(
-          sandbox.directories.map(
-            directory => directory.directoryShortid + directory.title
+const PreviewComponent: React.FC<Props> = ({
```

Comment:
```suggestion
export const Preview: React.FC<IPreviewProps> = ({
```

Just export the component directly, no need to declare the function then re-export it at the end of the file. I'm assuming the thought behind this is to ensure that the component has a display name in the React dev tools, for which we're using a Babel plugin that handles this automatically.

Reply:
Thank you. Updated. :)